### PR TITLE
typecheck: Handle ExtSlice in Subscript nodes

### PR DESCRIPTION
**Pull request by simeonkr**
I remember that we talked about there not being a need to visit `ExtSlice` nodes. I could move the code in `visit_extslice` to be handled as a case in `visit_subscript` if it's a problem that `ExtSlice` nodes are assigned types.

### CODE REVIEW

**Comment by simeonkr** [comment]
I'm looking into the failing test. It's very odd because it passes for me locally.

**Comment by david-yz-liu** [comment]
I reran the tests and they pass now. However, this implementation is too specific. A few comments:

1. `ExtSlice` is a general node that should support *arbitrary*, *multiple* expressions in the square brackets in a subscript node, e.g. `x[1, 2, 'hi', [int, str]]`. (E.g., think about/look up the implementation of the `List` of `Tuple` types from `typing`.
2. Looking at the implementation of these, you should typecheck `ExtSlice` by doing a `handle_call` against the `__getitem__` method. These methods take just a single argument other than `self`, and this argument is a *tuple* of values.
3. This means you can actually set the type of the `ExtSlice` node to be a `Tuple` (I didn't realize this before). Makes the `handle_call` unification easier.

File: python_ta/transforms/type_inference_visitor.py [hunk 1]
```diff
@@ -388,12 +388,17 @@ def visit_slice(self, node: astroid.Slice) -> None:
         if node.inf_type.getValue() == type(None):
             node.inf_type = TypeInfo(slice)
 
+    def visit_extslice(self, node: astroid.ExtSlice):
+        node.inf_type = TypeInfo(Tuple)
```
Comment (david-yz-liu) [inline python_ta/transforms/type_inference_visitor.py]: This is better, but I actually want you to get the individual elements as well (so the example below would have an `ExtSlice` of type `Tuple[int, str]`.

File: python_ta/transforms/type_inference_visitor.py [hunk 2]
```diff
@@ -388,12 +388,23 @@ def visit_slice(self, node: astroid.Slice) -> None:
         if node.inf_type.getValue() == type(None):
             node.inf_type = TypeInfo(slice)
 
+    def visit_extslice(self, node: astroid.ExtSlice):
+        dim_types = []
+        for dim in node.dims:
```
Comment (david-yz-liu) [inline python_ta/transforms/type_inference_visitor.py]: Seems like this is a perfect place for `collect_failables`

**Pull request merged**